kura/org.eclipse.kura.deployment.agent/src/main/java/org/eclipse/kura/deployment/agent/impl/DeploymentAgent.java

```diff
@@ -436,37 +437,45 @@ private File getFileFromFilesystem(URL url) {
 
     private void addPackageToConfFile(String packageName, String packageUrl) {
         Properties deployedPackages = readDeployedPackages();
+        Properties oldDeployedPackages = new Properties();
+        oldDeployedPackages.putAll(deployedPackages);
         deployedPackages.setProperty(packageName, packageUrl);
 
         if (this.dpaConfPath == null) {
             logger.warn("Configuration file not specified");
             return;
         }
 
-        try (FileOutputStream fos = new FileOutputStream(this.dpaConfPath)) {
-            deployedPackages.store(fos, null);
-            fos.flush();
-            fos.getFD().sync();
-        } catch (IOException e) {
-            logger.error("Error writing package configuration file", e);
+        if (!oldDeployedPackages.equals(deployedPackages)) {
+            try (FileOutputStream fos = new FileOutputStream(this.dpaConfPath)) {
+                deployedPackages.store(fos, null);
+                fos.flush();
+                fos.getFD().sync();
+            } catch (IOException e) {
+                logger.error("Error writing package configuration file", e);
+            }
         }
     }
 
     private void removePackageFromConfFile(String packageName) {
         Properties deployedPackages = readDeployedPackages();
+        Properties oldDeployedPackages = new Properties();
+        oldDeployedPackages.putAll(deployedPackages);
         deployedPackages.remove(packageName);
 
         if (this.dpaConfPath == null) {
             logger.warn("Configuration file not specified");
             return;
         }
 
-        try (FileOutputStream fos = new FileOutputStream(this.dpaConfPath)) {
-            deployedPackages.store(fos, null);
-            fos.flush();
-            fos.getFD().sync();
-        } catch (IOException e) {
-            logger.error("Error writing package configuration file", e);
+        if (!oldDeployedPackages.equals(deployedPackages)) {
+            try (FileOutputStream fos = new FileOutputStream(this.dpaConfPath)) {
+                deployedPackages.store(fos, null);
+                fos.flush();
+                fos.getFD().sync();
+            } catch (IOException e) {
+                logger.error("Error writing package configuration file", e);
+            }
         }
```

Comment:
Same as before: why not using an Early Return?

```suggestion
        if (oldDeployedPackages.equals(deployedPackages)) {
           return;
        }

        try (FileOutputStream fos = new FileOutputStream(this.dpaConfPath)) {
            deployedPackages.store(fos, null);
            fos.flush();
            fos.getFD().sync();
        } catch (IOException e) {
            logger.error("Error writing package configuration file", e);
        }
```